### PR TITLE
Update types

### PR DESCRIPTION
**Pull request by mattludwigs**
Provide some type alias to help provide type meaning.

### CODE REVIEW

File: lib/i2c.ex [hunk 1]
```diff
@@ -31,7 +35,7 @@ defmodule Circuits.I2C do
   On success, this returns a reference to the I2C bus resource.  Use the
   reference in subsequent calls to read/write I2C device
   """
-  @spec open(binary) :: {:ok, reference} | {:error, term}
+  @spec open(i2c_device :: binary) :: {:ok, reference} | {:error, term}
```
Comment (fhunleth) [inline lib/i2c.ex]: screen shot 2018-11-04 at 7 04 42 pm

That's what the docs look like. If I get the gist of what you want to achieve, how do you feel about renaming the parameter (not the type, but the actual parameter to make it more clear). Also, I was thinking that the reference returned deserved a type of its own. Plus parentheses to make them consistent with ex_doc. How's this:

```elixir
@spec open(String.t()) :: {:ok, i2c_device()} | {:error, any()}
def open(device_name) do
...
```

Comment (fhunleth) [reply lib/i2c.ex]: I know this parentheses thing is going to cause pain. My feeling is that the battle has been lost. ex_doc does parentheses, the Visual Studio Code plugin (and I assume others) puts in parentheses by default. I realize there are exceptions in the Elixir code base and if you bring this up with José and he says "no parentheses" is the Elixir convention and ex_doc is wrong, then I'll be good with no parentheses.

Comment (fhunleth) [reply lib/i2c.ex]: I thought about this some more. What I wrote sounds like a higher bar than I intended. I'm fine with punting for now, but we should file an issue on the Elixir project on parentheses vs. no parentheses to confirm or refute the direction that ex_doc has taken.

Comment (mattludwigs) [reply lib/i2c.ex]: I can update, I don't mind.  I like that reference being a type of its own actually, I did not think of that.

**Comment by fhunleth** [comment]
@mattludwigs Thanks for the updates. One big observation that I had from them is that I2C buses were being referred to as devices. This was a huge oversight on my part, so I'm super thankful that updating the typespecs made this obvious. I amended your commit with that change. Feel free to send a follow-up PR if I messed up.

I also added dialyxir to the project so that we could run it on CircleCI to validate the specs.

Last, I removed the `bytes_to_read` spec. For some reason I thought it was more clear to see the variable named "bytes_to_read" and then have the spec on the line before say pos_integer(). As it was, you had to go to the top of the file to see that it was expecting a positive integer. Anyway, I hope you don't mind that change.

**Pull request merged**